### PR TITLE
Add evm hash and flatten evm_log

### PR DESCRIPTION
**Pull request by florian-stoica-tlabs**

### CODE REVIEW

File: substrate-archive/src/migrations/v5/1641997527150-StoreEvmLog.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        await queryRunner.query(`
+            CREATE OR REPLACE FUNCTION json_arr2text_arr(_js jsonb)
+              RETURNS text[] LANGUAGE sql IMMUTABLE PARALLEL SAFE AS
+            'SELECT ARRAY(SELECT jsonb_array_elements_text(_js))';
+        `)
+
+        await queryRunner.query(
+            `ALTER TABLE "substrate_event" ADD COLUMN IF NOT EXISTS "evm_log_topics"  TEXT [] GENERATED ALWAYS AS (case when name = 'evm.Log' then json_arr2text_arr(params->0->'value'->'topics') end) STORED;`
+        )
```
Comment (dzhelezov) [inline substrate-archive/src/migrations/v5/1641997527150-StoreEvmLog.ts:21]: Do we need to add indices here to make sure filtering works fast?

File: substrate-archive-gateway/metadata/databases/HydraIndexer/tables/public_substrate_event.yaml [hunk 2]
```diff
@@ -13,6 +13,10 @@ configuration:
     extrinsic_index: extrinsicIndex
     extrinsic_name: extrinsicName
     index_in_block: indexInBlock
+    evm_log_address: evmLogAddress
```
Comment (dzhelezov) [inline substrate-archive-gateway/metadata/databases/HydraIndexer/tables/public_substrate_event.yaml:16]: Are we going to have these columns for all archives or only evm supported?

**Review by dzhelezov — COMMENTED** [review]
Great work! The only concern I have is what is the plan to keep the EVM and non-EVM stuff separated, as clearly evm-specific columns and handlers make no sense for the chains without the evm support.

File: substrate-archive-gateway/metadata/databases/HydraIndexer/tables/public_substrate_event.yaml (hunk 2 above)
Comment (eldargab) [reply substrate-archive-gateway/metadata/databases/HydraIndexer/tables/public_substrate_event.yaml:16]: Definitely only for evm.

Comment (eldargab) [reply substrate-archive-gateway/metadata/databases/HydraIndexer/tables/public_substrate_event.yaml:16]: Let's keep archive changes out of this repo for the moment.

File: substrate-processor/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,2751 @@
+{
```
Comment (eldargab) [inline substrate-processor/package-lock.json]: Ups, please use rush :)

File: substrate-processor/src/batch.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ export interface DataHandlers {
      * Mapping of type `trigger event` -> `extrinsic` -> `extrinsic handler list`
      */
     extrinsics: Record<QualifiedName, Record<QualifiedName, ExtrinsicHandler[]>>
+    evmLogs: Record<QualifiedName, EvmLogHandler[]>
```
Comment (eldargab) [inline substrate-processor/src/batch.ts:17]: `QualifiedName` has a meaning of `${section}.${item}`. Contract address definitely falls outside of this category, please create a separate alias for that.

File: substrate-processor/src/ingest.ts [hunk 5]
```diff
@@ -136,6 +137,9 @@ export class Ingest {
             events.forEach(name => {
                 or.push(`events: {_contains: [{name: "${name}"}]}`)
             })
+            if (evmLogs && evmLogs.length > 0) {
```
Comment (eldargab) [inline substrate-processor/src/ingest.ts:140]:
```ts
evmLogs?.length
```

File: substrate-processor/src/interfaces/handlerContext.ts [hunk 6]
```diff
@@ -28,6 +28,14 @@ export interface EventHandler {
 }
 
 
+export interface EvmLogHandlerContext extends EventHandlerContext {
```
Comment (eldargab) [inline substrate-processor/src/interfaces/handlerContext.ts:31]: Should be expanded. The suggestion is to keep the existing `SubstrateBlock`, `SubstrateEvent` interfaces intact.
But, evm specific handlers we'll get additional data.

**Review by eldargab — CHANGES_REQUESTED** [review]

**Pull request merged**